### ISSUE TITLE
Missing Dependency "goog.string" not found even for the test app in the repo.

### ISSUE THREAD

**Issue by thefunnyguy** [karma-runner/karma-closure#15]
I am trying to make karma closure plugin work for me.
I did all the changes mentioned in the repo.

But i am unable to resolve the dependencies for the goog.

So i cloned the whole repo and trying running the karma start test-app/karma.config.js
it doesn't find "goog.string" Shows the error here:
https://github.com/karma-runner/karma-closure/blob/master/lib/resolver.js#L37

**Comment by moderndegree** [comment]
Any word on this?

**Comment by moderndegree** [comment]
Apparently, this is an issue with new versions of node. I am running v0.12.0 and am seeing the same issue. A co-worker is running v0.10.26 and doesn't have the issue. I have been able to determine that the problem is being caused by line #34 of `lib/resolver.js`. It is looking for `provideMap[dep]`, which is not there. The path, is actually `providerMap.__proto__[dep]`. I resolved this by adding the following:

```
else if (provideMap.__proto__[dep]) {
    resolveFile(provideMap.__proto__[dep], files, alreadyResolvedMap);
}
```

I'll do some digging to see why this is happening but it has unblocked me for the time being.

**Comment by mohannarang** [comment]
Replacing
  var fileMap = Object.create(null);
  var provideMap = Object.create(null);

with

  var fileMap = Object.create({});
  var provideMap = Object.create({});

Worked for me. Passing null breaks the proto chain but {} maintains it.

**Comment by moderndegree** [comment]
Nice. I was just reading about that. Thanks. Have you submitted a pull request?

**Comment by moderndegree** [comment]
nm... it looks like someone already did but it appears to have broken the build.

**Comment by mohannarang** [comment]
I created one long time back ... but it was failing the build. So i cloned the repo and made the changes there and used it.
https://github.com/bhojo/karma-closure

**Comment by mlippert** [comment]
I was running into this issue also. I resolved it by replacing the explicit references to `__proto__` with `Object.setPrototypeOf` in the `loadExternalDeps` function:

```javascript
    //fileMap.__proto__ = parsed.fileMap;
    //provideMap.__proto__ = parsed.provideMap;
    Object.setPrototypeOf(fileMap, parsed.fileMap);
    Object.setPrototypeOf(provideMap, parsed.provideMap);
```

**Comment by benjarwar** [comment]
Just ran into this issue, and the proposed fix by @mlippert does the trick. Any news on getting this merged in?

**Comment by scottsandersdev** [comment]
Resolved in https://github.com/karma-runner/karma-closure/commit/e0b15eacc69aacd9418c580856773d72eeaa496f. Released in v0.1.3
